woocommerce/woocommerce-ios
# Product selector: Add Clear selection button

⚠️ Please make sure that #6696 has been reviewed ⚠️

Part of #6489 


### Description

This PR completes the updated product selector with a new button to clear selected items:
- "Clear selection" is only available if any item in the product list is selected.
- Enable the Done button when no item is selected to make sure there's a way to submit the selection without any selected item.
- Previously in #6679 we had to set up initially selected items after the product list is updated - to make sure that paginated items are pre-selected properly. There's an edge case: if any item is unselected manually, reloading the product list with re-select the item automatically. I've added a fix to update `initialSelectedItems` to mitigate this.

### Testing instructions

Testing coupon edit and coupon restrictions:
- Make sure that your test store has coupons enabled and has set up at least one coupon in WP Admin > Marketing > Coupons.
- Open the app, make sure that coupon management is enabled in Menu > Settings > Experimental Features > Coupon Management.
- Navigate to Menu > Coupons and select a coupon.
- Tap the ellipsis button > Edit Coupon > Select the All Products/Edit Products button.
- Notice that the button "Clear selection" shows up only when selecting any items on the product list.
- Tap the "Clear selection" button, notice that any previously selected items are all toggled.
- Select some items, then submit the selection. Open the product selector again and immediately tap "Clear selection". If there are more than 25 items in the list, scroll down to next pages to see that the previously selected items are not automatically selected again.

Please feel free to test the product selector on the New Order screen to make sure that single selection still works as before.

### Screenshots


https://user-images.githubusercontent.com/5533851/165246997-c8fc28fd-45a5-449c-9b7e-855722dbf07a.mp4



---
